chat_shell/chat_shell/tools/builtin/canvas_artifact.py

```
@@ -0,0 +1,469 @@
+    """Input schema for updating an existing artifact."""
+
+    artifact_id: str = Field(description="ID of the artifact to update")
+    content: str = Field(description="Updated content")
+    title: str | None = Field(default=None, description="New title (optional)")
+
+    @field_validator("artifact_id")
+    @classmethod
+    def validate_artifact_id(cls, v: str) -> str:
+        """Validate artifact_id is a valid UUID."""
+        if not v or not v.strip():
+            raise ValueError("Artifact ID cannot be empty")
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)